packages/core/src/package/packageQuery/PackageFetcher.ts

```diff
@@ -0,0 +1,29 @@
+import { Connection } from "@salesforce/core";
+import QueryHelper from "../../queryHelper/QueryHelper";
+
+
+const query =
+  'SELECT Id,Name, Description, NamespacePrefix, ContainerOptions, IsOrgDependent ' +
+  'FROM Package2 ' +
+  'WHERE IsDeprecated != true ' +
+  'ORDER BY NamespacePrefix, Name';
+
+
+export default class PackageFetcher
+{
+
+  
+  constructor(
+    private conn: Connection
+  ) {}
+
+  public async listAllPackages()
+  {
+    let records = await QueryHelper.query<any>(query, this.conn, true);
```

Comment:
Record type should be provided as type parameter, instead of `any`, so that the accessible fields on the `Package2` object are known.